### PR TITLE
Models in game

### PR DESCRIPTION
**Pull request by Eve0028**

### CODE REVIEW

File: rpgame/src/main/java/pk/rpgame/view/PrintEventVisitor.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import pk.rpgame.model.state.ActiveRoomState;
+import pk.rpgame.model.state.RoomState;
+import pk.rpgame.model.state.VisitedRoomState;
+
+public class PrintEventVisitor implements Visitor {
+    public void visitRoom(Room room) {
+        if (room.getState().getClass() == ActiveRoomState.class) {
+            System.out.println("You entered " + room);
+            //TODO
+            // Change color of the room - I'm not sure if that gonna be here
```
Comment (Kaponn) [inline rpgame/src/main/java/pk/rpgame/view/PrintEventVisitor.java]: I think it should, color of the room should change right after visitor enters or leaves it

File: rpgame/src/main/java/pk/rpgame/view/RoomVisualisation.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        this.charRoom = charRoom;
+    }
+
+    public String getColor() {
+        return colorStr;
+    }
+    public Character getChar() {
+        return charRoom;
+    }
+}
```
Comment (Kaponn) [inline rpgame/src/main/java/pk/rpgame/view/RoomVisualisation.java]: no newline at the end of file

**Review by Kaponn — APPROVED** [review]

File: rpgame/src/main/java/pk/rpgame/model/items/Potion.java [hunk 3]
```diff
@@ -19,4 +18,10 @@ public double getHealth() {
     public void use(LivingEntity entity) {
         entity.setHealth(entity.getHealth() + this.health);
     }
+
+    @Override
+    public String toString() {
+        return super.toString()
+                + " [health: " + this.getHealth() + "]";
```
Comment (Kaponn) [inline rpgame/src/main/java/pk/rpgame/model/items/Potion.java:31]: i think we should have one convention for this functions, either like here "[health: sth]", or like in armor "\narmor: sth"

File: rpgame/src/main/java/pk/rpgame/model/items/Weapon.java [hunk 4]
```diff
@@ -4,7 +4,24 @@
...
+    }
+
+    public void setAttack(double attack) {
+        this.attack = attack;
+    }
+
+    @Override
+    public String toString() {
+        return super.toString()
+                + "\nattack damage: " + this.getAttack();
```
Comment (Kaponn) [inline rpgame/src/main/java/pk/rpgame/model/items/Weapon.java]: same as with health

File: rpgame/src/main/java/pk/rpgame/view/PrintEventVisitor.java (hunk 1 above)
Comment (Eve0028) [reply rpgame/src/main/java/pk/rpgame/view/PrintEventVisitor.java]: I passed this task to LevelMap, which informs about changing rooms to the Map using Observer

**Pull request merged**